From d96f1f13a0d4f8815643b9b4a6e6bf0c13ccf0bc Mon Sep 17 00:00:00 2001
From: "Yan, WalterX" <walterx.yan@intel.com>
Date: Fri, 27 Oct 2017 15:44:01 +0800
Subject: [PATCH 01/40] Only hide usb notificaiton when it's in host mode.

For the devices have female usb interface and some usb
devices plugged-in, e.g. mouse, keyboard, the usb notification
will be hidden incorrectly while connecting as a client.

To scan the usb devices only when it's in host mode.

Change-Id: I1c648f597fa7154d1a44b40ff831652730e819cd
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-58562
Signed-off-by: Yan, WalterX <walterx.yan@intel.com>
---
 .../android/server/usb/UsbDeviceManager.java  | 36 ++++++++++---------
 1 file changed, 19 insertions(+), 17 deletions(-)

diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 36f8063cca52..3510d5f14396 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -877,23 +877,25 @@ public class UsbDeviceManager implements ActivityManagerInternal.ScreenObserver
                     }
 
                     mHideUsbNotification = false;
-                    while (devices.hasNext()) {
-                        Map.Entry pair = (Map.Entry) devices.next();
-                        if (DEBUG) {
-                            Slog.i(TAG, pair.getKey() + " = " + pair.getValue());
-                        }
-                        UsbDevice device = (UsbDevice) pair.getValue();
-                        int configurationCount = device.getConfigurationCount() - 1;
-                        while (configurationCount >= 0) {
-                            UsbConfiguration config = device.getConfiguration(configurationCount);
-                            configurationCount--;
-                            int interfaceCount = config.getInterfaceCount() - 1;
-                            while (interfaceCount >= 0) {
-                                UsbInterface intrface = config.getInterface(interfaceCount);
-                                interfaceCount--;
-                                if (sBlackListedInterfaces.contains(intrface.getInterfaceClass())) {
-                                    mHideUsbNotification = true;
-                                    break;
+                    if (mHostConnected) {
+                        while (devices.hasNext()) {
+                            Map.Entry pair = (Map.Entry) devices.next();
+                            if (DEBUG) {
+                                Slog.i(TAG, pair.getKey() + " = " + pair.getValue());
+                            }
+                            UsbDevice device = (UsbDevice) pair.getValue();
+                            int configurationCount = device.getConfigurationCount() - 1;
+                            while (configurationCount >= 0) {
+                                UsbConfiguration config = device.getConfiguration(configurationCount);
+                                configurationCount--;
+                                int interfaceCount = config.getInterfaceCount() - 1;
+                                while (interfaceCount >= 0) {
+                                    UsbInterface intrface = config.getInterface(interfaceCount);
+                                    interfaceCount--;
+                                    if (sBlackListedInterfaces.contains(intrface.getInterfaceClass())) {
+                                        mHideUsbNotification = true;
+                                        break;
+                                    }
                                 }
                             }
                         }
-- 
2.17.1

